### PR TITLE
feat: add Ingress integration

### PR DESCRIPTION
**Pull request by amandahla**
Applicable spec: <link>

### Overview

Add Ingress integration to Maubot charm.

### Rationale

Allow external access.

### Juju Events Changes

ingress ready
ingress revoked

### Library Changes

Add charms.traefik_k8s.v2.ingress

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [ ] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

No CH available yet.

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -51,6 +59,24 @@ def _on_config_changed(self, _: ops.ConfigChangedEvent) -> None:
         container.replan()
         self.unit.status = ops.ActiveStatus()
 
+    def _on_ingress_ready(self, _: IngressPerAppReadyEvent) -> None:
+        """Handle ingress ready event."""
+        container = self.unit.get_container(MAUBOT_CONTAINER_NAME)
+        if not container.can_connect():
+            return
+        container.add_layer(MAUBOT_CONTAINER_NAME, self._pebble_layer, combine=True)
```
Comment (javierdelapuente) [inline src/charm.py:67]: in the function `_on_config_changed`, the layer is added with: `container.add_layer(MAUBOT_SERVICE_NAME, self._pebble_layer, combine=True)`. Both constants, MAUBOT_CONTAINER_NAME and MAUBOT_SERVICE_NAME are currently the same string, but I think it is confusing.

**Review by javierdelapuente — DISMISSED** [review]
Looks good. 

I was going to suggest to extract a "reconcile" function, but I see that it is ready in another PR!.

Just a small comment. I approved anyway , as I see that it is also tackled in the other PR.

File: tests/integration/test_charm.py [hunk 2]
```diff
@@ -29,5 +29,17 @@ async def test_build_and_deploy(
...
+        channel="edge",
+        config={
+            "path-routes": "/",
+            "service-hostname": "maubot.local",
+            "service-namespace": ops_test.model.name,
+            "service-name": "maubot",
+        },
+        trust=True,
+    )
+    await ops_test.model.add_relation(maubot.name, nginx_ingress_integrator.name)
```
Comment (cbartz) [inline tests/integration/test_charm.py:44]: Is there a query against the ingress IP that we can use to test the working ingress?

**Review by cbartz — APPROVED** [review]
LGTM, have a suggestion for testing

**Review by javierdelapuente — APPROVED** [review]

File: tests/integration/test_charm.py (hunk 2 above)
Comment (amandahla) [reply tests/integration/test_charm.py:44]: For now, the integration test only validates that the application is active after integrating with NGINX.
I'll add a more elaborate integration test once we have the configuration and the matrix-plugin in place.

**Pull request merged**